### PR TITLE
Document flakiness of WallClock tests

### PR DESCRIPTION
**Pull request by amesgen**
# Description

Reported by @coot (found in https://github.com/input-output-hk/ouroboros-network/pull/3785#issuecomment-1180780005), thanks!

# Checklist

- Branch
    - [x] Commit sequence broadly makes sense
    - [x] Commits have useful messages
    - [x] New tests are added if needed and existing tests are updated
    - [x] If this branch changes Consensus and has any consequences for downstream repositories or end users, said changes must be documented in [`interface-CHANGELOG.md`](../ouroboros-consensus/docs/interface-CHANGELOG.md)
    - [x] If this branch changes Network and has any consequences for downstream repositories or end users, said changes must be documented in [`interface-CHANGELOG.md`](../docs/interface-CHANGELOG.md)
    - [x] If serialization changes, user-facing consequences (e.g. replay from genesis) are confirmed to be intentional.
- Pull Request
    - [x] Self-reviewed the diff
    - [x] Useful pull request description at least containing the following information:
      - What does this PR change?
      - Why these changes were needed?
      - How does this affect downstream repositories and/or end-users?
      - Which ticket does this PR close (if any)? If it does, is it [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)?
    - [x] Reviewer requested

### CODE REVIEW

File: ouroboros-consensus-test/test-consensus/Test/Consensus/BlockchainTime/Simple.hs [hunk 1]
```diff
@@ -288,8 +289,10 @@ prop_delayNoClockShift =
                    (slotLengthFromMillisec 100)
                    (secondsToNominalDiffTime 20)
                    5
-      assertEqual "slots" slots [SlotNo n | n <- [0..4]]
+      pure $ slots === [SlotNo n | n <- [0..4]]
 
+-- | Note that that under load, the returned list could be missing certain slots
```
Comment (dnadales) [inline ouroboros-consensus-test/test-consensus/Test/Consensus/BlockchainTime/Simple.hs:294]: Do we need to prepend this with a `TODO` or `FIXME`?

File: ouroboros-consensus-test/test-consensus/Test/Consensus/BlockchainTime/Simple.hs [hunk 2]
```diff
@@ -85,15 +84,17 @@ prop_delayNextSlot :: TestDelayIO -> Property
...
+    test :: IO Property
     test = do
         tdioStart  <- pickSystemStart
         let time = defaultSystemTime tdioStart nullTracer
         atStart    <- fst <$> getWallClockSlot  time tdioSlotLen
         nextSlot   <-         waitUntilNextSlot time tdioSlotLen maxClockRewind atStart
         afterDelay <- fst <$> getWallClockSlot  time tdioSlotLen
-        assertEqual "atStart + 1" (atStart + 1) afterDelay
-        assertEqual "nextSlot"    nextSlot      afterDelay
+        pure $ conjoin
```
Comment (dnadales) [inline ouroboros-consensus-test/test-consensus/Test/Consensus/BlockchainTime/Simple.hs:94]: :ok_hand:

**Review by dnadales — APPROVED** [review]
Thanks!

File: ouroboros-consensus-test/test-consensus/Test/Consensus/BlockchainTime/Simple.hs (hunk 1 above)
Comment (amesgen) [reply ouroboros-consensus-test/test-consensus/Test/Consensus/BlockchainTime/Simple.hs:294]: I added a link to the tech debt issue: https://github.com/input-output-hk/ouroboros-network/compare/3fb8d691f2ed562484b673f9f9c55a4fb74e3486..f65d9093e8fa2e1284bf3a0e4bfc6b9aef2f70f3

**Comment by amesgen** [comment]
bors merge

**Pull request merged**